olive/strategy/search_strategy.py

```diff
@@ -120,13 +120,15 @@ def _next_search_group(self, init_model_id: Optional[str] = None) -> Optional[Se
         """
         Get the next search space group and initialize the search algorithm.
         """
-        # TODO: organize the state better and make exection order more flexible using a graph
+        # TODO: organize the state better and make execution order more flexible using a graph
         if self._active_spaces_group is not None:
             self._done_spaces_groups.append(self._active_spaces_group)
             # legacy, will update once search results has info function
             sorted_model_ids, sorted_search_points, sorted_results = self._search_results[
                 tuple(self._active_spaces_group)
-            ].sort_search_points()
+            ].sort_search_points(apply_goals=True)
+            # TODO: this is a hack to get the best search point for the current search space group
+            #      it totally work for joint execution order, but not for pass-by-pass
```

Comment:
Does this mean we don't support `pass-by-pass` for now?

Reply:
For multi-output during the pass running, it does. We just put the top one after sorting the results in pass-by-pass. But you know, for multi-objective optimization, there are several available models to feed into next pass.